### ISSUE TITLE
flutter run: Timeout of 120000 reached waiting for exclusive access to file

### ISSUE THREAD

**Issue by jaitjacob** [flutter/flutter#45037]
## Steps to Reproduce

1. fresh installation of flutter
2. run flutter doctor and ensure "no issues found" is shown
3. run fluter run

**Target Platform:  Android**
**Target OS version/browser: Android Pi API 28**
**Devices: Haven't tried running on an emulator or simulator yet.**

## Crash report

Flutter crash report; please file at https://github.com/flutter/flutter/issues.

### command
`flutter run`

### exception
ProcessException: ProcessException: Process "C:\Users\jaitj\Documents\Summer 2019\flutter\myapp\android\gradlew.bat" exited abnormally:
Exception in thread "main" java.lang.RuntimeException: Timeout of 120000 reached waiting for exclusive access to file: C:\Users\jaitj\.gradle\wrapper\dists\gradle-4.10.2-all\9fahxiiecdb76a5g3aw9oi8rv\gradle-4.10.2-all.zip
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:61)
	at org.gradle.wrapper.Install.createDist(Install.java:48)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
  Command: C:\Users\jaitj\Documents\Summer 2019\flutter\myapp\android\gradlew.bat -v

```
#0      runCheckedAsync (package:flutter_tools/src/base/process.dart:259:7)
<asynchronous suspension>
#1      _initializeGradle (package:flutter_tools/src/android/gradle.dart:300:9)
<asynchronous suspension>
#2      _ensureGradle (package:flutter_tools/src/android/gradle.dart:281:37)
<asynchronous suspension>
#3      _readGradleProject (package:flutter_tools/src/android/gradle.dart:192:31)
<asynchronous suspension>
#4      _gradleAppProject (package:flutter_tools/src/android/gradle.dart:112:37)
<asynchronous suspension>
#5      getGradleAppOut (package:flutter_tools/src/android/gradle.dart:106:29)
<asynchronous suspension>
#6      AndroidApk.fromAndroidProject (package:flutter_tools/src/application_package.dart:164:23)
<asynchronous suspension>
#7      ApplicationPackageFactory.getPackageForPlatform (package:flutter_tools/src/application_package.dart:46:32)
<asynchronous suspension>
#8      FlutterDevice.runHot (package:flutter_tools/src/resident_runner.dart:359:56)
<asynchronous suspension>
#9      HotRunner.run (package:flutter_tools/src/run_hot.dart:254:39)
<asynchronous suspension>
#10     RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:472:37)
<asynchronous suspension>
#11     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:490:18)
#12     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#13     _rootRunUnary (dart:async/zone.dart:1132:38)
#14     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#15     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#16     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#17     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#18     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#19     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
#20     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
#21     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#22     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#23     _rootRunUnary (dart:async/zone.dart:1132:38)
#24     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#25     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#26     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#27     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#28     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#29     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
#30     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
#31     AndroidDevice.isLocalEmulator (package:flutter_tools/src/android/android_device.dart)
#32     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#33     _rootRunUnary (dart:async/zone.dart:1132:38)
#34     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#35     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#36     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#37     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#38     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#39     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
#40     _rootRun (dart:async/zone.dart:1124:13)
#41     _CustomZone.run (dart:async/zone.dart:1021:19)
#42     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#43     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#44     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#45     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#46     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#47     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```

## flutter analyze

```
Analyzing myapp...
No issues found! (ran in 1.4s)
```

## Flutter doctor 

```
C:\Users\jaitj\Documents\Summer 2019\flutter\myapp>flutter doctor -v
[√] Flutter (Channel stable, v1.9.1+hotfix.6, on Microsoft Windows [Version
    10.0.18362.476], locale en-IN)
    • Flutter version 1.9.1+hotfix.6 at
      C:\src\flutter_windows_v1.9.1+hotfix.6-stable\flutter
    • Framework revision 68587a0916 (9 weeks ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\jaitj\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling      
      support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)     
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)     

[√] VS Code (version 1.40.1)
    • VS Code at C:\Users\jaitj\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.6.0

[√] Connected device (1 available)
    • Mi A1 • 7324ca9d0504 • android-arm64 • Android 9 (API 28)

• No issues found!
```

**Comment by jaitjacob** [comment]
After a restart `flutter run` gives me the following output
`>flutter run
Launching lib\main.dart on Mi A1 in debug mode...
Initializing gradle...                                              1.5s
Resolving dependencies...                                          19.5s

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':app:_internal_aapt2_binary'.
> Could not resolve com.android.tools.build:aapt2:3.2.1-4818971.
  Required by:
      project :app
   > Could not resolve com.android.tools.build:aapt2:3.2.1-4818971.
      > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/build/aapt2/3.2.1-4818971/aapt2-3.2.1-4818971.pom'.
         > Could not GET 'https://dl.google.com/dl/android/maven2/com/android/tools/build/aapt2/3.2.1-4818971/aapt2-3.2.1-4818971.pom'.
            > dl.google.com
   > Could not resolve com.android.tools.build:aapt2:3.2.1-4818971.
      > Could not get resource 'https://jcenter.bintray.com/com/android/tools/build/aapt2/3.2.1-4818971/aapt2-3.2.1-4818971.pom'.
         > Could not GET 'https://jcenter.bintray.com/com/android/tools/build/aapt2/3.2.1-4818971/aapt2-3.2.1-4818971.pom'.
            > jcenter.bintray.com

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
Running Gradle task 'assembleDebug'...
Running Gradle task 'assembleDebug'... Done                        17.1s`

**Comment by kassadin** [comment]
Failure caused by slow network connection, you may need to use a proxy.

**Comment by janmoppel** [comment]
Hello @jaitjacob !

Thank you for posting an issue.
I agree with @kassadin, it seems like a network problem. Can you please check whether application will run in other network?

**Comment by jaitjacob** [comment]
Hey @janmoppel! Thanks for reaching out. So I was running my laptop on my home WiFi. Switched to mobile data now ironically seems to work fine. 

Here is some more info if you'd like to see it.
```

C:\Users\jaitj\Documents\Summer 2019\flutter\myapp>flutter run
Waiting for another flutter command to release the startup lock...
Launching lib\main.dart on Mi A1 in debug mode...
Initializing gradle...                                              2.6s
Resolving dependencies...                                          27.7s
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                        81.8s
Built build\app\outputs\apk\debug\app-debug.apk.
Installing build\app\outputs\apk\app.apk...                         6.2s
I/OpenGLRenderer(28842): Initialized EGL, version 1.4
D/OpenGLRenderer(28842): Swap behavior 2
I/Choreographer(28842): Skipped 131 frames!  The application may be doing too much work on its main thread.
D/vndksupport(28842): Loading /vendor/lib64/hw/android.hardware.graphics.mapper@2.0-impl.so from current namespace instead of sphal namespace.
D/vndksupport(28842): Loading /vendor/lib64/hw/gralloc.msm8953.so from current namespace instead of sphal namespace.
I/OpenGLRenderer(28842): Davey! duration=2266ms; Flags=1, IntendedVsync=172428291375061, Vsync=172430474708307, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=172430479692187, AnimationStart=172430480143124, PerformTraversalsStart=172430480156093, DrawStart=172430492237968, SyncQueued=172430494348384, SyncStart=172430494458489, IssueDrawCommandsStart=172430494732343, SwapBuffers=172430555313801, FrameCompleted=172430558179478, DequeueBufferDuration=3229000, QueueBufferDuration=691000,
Syncing files to device Mi A1...
 3,564ms (!)

�🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
An Observatory debugger and profiler on Mi A1 is available at: http://127.0.0.1:54875/rNxqTu8ysu0=/
For a more detailed help message, press "h". To detach, press "d"; to quit, press "q".



```

**Comment by iapicca** [comment]
Hi @jaitjacob 
given your last message I feel safe to close this issue,
if you disagree please write in the comments
and I will reopen it. 
Thank you
